core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -61,11 +61,14 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     //this assertion is only used for creating the metrics for DelayedFetchMetrics, it should never fail, but should not be removed
     assertNotNull(DelayedFetchMetrics)
 
-    val countOfStaticMetrics = Metrics.defaultRegistry.allMetrics.keySet.size
+    val originalMetrics = Metrics.defaultRegistry.allMetrics.keySet.asScala
 
     for (i <- 0 to 5) {
       createAndShutdownStep(topic, "group" + i % 3, "consumer" + i % 2, "producer" + i % 2)
-      assertEquals(countOfStaticMetrics, Metrics.defaultRegistry.allMetrics.keySet.size)
+      val missingMetricOpt = originalMetrics.filterNot(Metrics.defaultRegistry.allMetrics.keySet.asScala).headOption
+      val newMetricOpt = Metrics.defaultRegistry.allMetrics.keySet.asScala.filterNot(originalMetrics).headOption
```

Comment:
Would it be better to keep the sequence instead of getting the head?

Reply:
Yeah, I think it's OK to print them all in the worst case. In the common case, there would be a small number of metrics there.

Reply:
@ijuma Hmm.. this test is removed in commit 7132a85fc394bc0627fe1763c17cb523d8a8ff37 because it was deprecated. Should we add the test back?

Reply:
Good point. We probably don't need it.

Reply:
Sure. I will close the PR.